Include/setobject.h

```diff
@@ -67,10 +67,19 @@ typedef struct {
 #define PySet_GET_SIZE(so) (assert(PyAnySet_Check(so)),(((PySetObject *)(so))->used))
 
 PyAPI_DATA(PyObject *) _PySet_Dummy;
+PyAPI_DATA(PyTypeObject) _PyReproducibleFrozenSet_Type;
+
+PyAPI_FUNC(PyObject *) _PyReproducibleFrozenSet_New(PyObject *);
 
 PyAPI_FUNC(int) _PySet_NextEntry(PyObject *set, Py_ssize_t *pos, PyObject **key, Py_hash_t *hash);
 PyAPI_FUNC(int) _PySet_Update(PyObject *set, PyObject *iterable);
 
+#define _PyReproducibleFrozenSet_CheckExact(ob) \
+    Py_IS_TYPE(ob, &_PyReproducibleFrozenSet_Type)
+#define _PyReproducibleFrozenSet_Check(ob) \
+    (Py_IS_TYPE(ob, &_PyReproducibleFrozenSet_Type) || \
+      PyType_IsSubtype(Py_TYPE(ob), &_PyReproducibleFrozenSet_Type))
+
```

Comment:
These functions should not be public. I think we really should not expose this type to be used or checked from outside CPython